sound/soc/sof/intel/hda.c

```diff
@@ -549,7 +440,7 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 
 	dev_dbg(sdev->dev, "using IPC IRQ %d\n", sdev->ipc_irq);
 	ret = request_threaded_irq(sdev->ipc_irq, hda_dsp_ipc_irq_handler,
-				   chip->ops->irq_thread, IRQF_SHARED,
+				   sdev->ops->irq_thread, IRQF_SHARED,
```

Comment:
I checked if we still need this chip information, and it seems we have another level of indirection with the hda stuff? can we simplify all this so that the chip info is retrieved in the same way across the code?

if (sdev->hda)
		chip = sdev->hda->desc;

Reply:
in this function: 	
        chip = get_chip_info(sdev->pdata);
        sdev->hda = hdev;
	hdev->desc = chip;

so it is fine here.

Reply:
not really. why do we need to tie ops to hda?

Reply:
Is ops tied to hda ?  ops is tied to sdev, and there is a reference in hda->chip_info->ops.